flutter_appauth/ios/Classes/FlutterAppauthPlugin.m

```diff
@@ -189,10 +193,18 @@ - (void)performAuthorization:(OIDServiceConfiguration *)serviceConfiguration cli
                 [self finishWithError:AUTHORIZE_ERROR_CODE message:[self formatMessageWithError:AUTHORIZE_ERROR_MESSAGE_FORMAT error:error] result:result];
             }
         }];
-        
     }
 }
 
+- (NSObject<OIDExternalUserAgent> *)userAgentWithEphemeralSession:(BOOL)preferEphemeralSession inRootViewController:(UIViewController *)rootViewController {
```

Comment:
Are you able to come up with a better name for this method? Currently it implies it returns a user agent with an ephemeral session. The only idea I have at the moment is `initUserAgent`

Reply:
I struggled with this name as well... I was trying to jump back into my old OjbC days where the method name is meant to read like documentation. I've decided to change it to
`- (NSObject<OIDExternalUserAgent> *)userAgentWithViewController:(UIViewController *)rootViewController useEphemeralSession:(BOOL)useEphemeralSession
`

Since creating a userAgent always requires a VC, that is moved to the first argument. I think that makes it more clear that useEphemeralSession is option you are choosing whether to enable or not.

To me, `init*` in this case seems incorrect since it's not part of a class wide setup.